### PR TITLE
gettype2 prototype implementation

### PR DESCRIPTION
**Pull request by krux02**
As already mentioned, this gettype2 implementation does not exactly do what it is supposed to do, because I could not find a nice way that would work in all cases without destroying other cases. The idea is that `gettype2` and `getimpl` always return something that resembles the left of the right side of the the type assignment. for types that are builtin the result of `getimpl` is unspecified.

### CODE REVIEW

**Comment by Araq** [comment]
https://travis-ci.org/nim-lang/Nim/builds/101966218 you even broke booting ;-)

File: compiler/vmdeps.nim [hunk 1]
```diff
@@ -181,3 +181,65 @@ proc mapTypeToAst(t: PType, info: TLineInfo; allowRecursion=false): PNode =
...
+      let void = atomicTypeX("void", t, info)
+      void.typ = newType(tyEmpty, t.owner)
+      result.add void
+    else:
+      result.add opMapTypeToAst2(t.sons[i], info)
+
+proc opMapTypeToAst2(t: PType; info: TLineInfo; getImpl: bool): PNode =
+
+  proc name(tk : TTypeKind): string =
+    if tk == tySequence:
```
Comment (Araq) [inline compiler/vmdeps.nim]: Better rename 'tySequence' in the compiler everywhere finally.

**Comment by krux02** [comment]
Yes, I introduced several magics, and added some code duplication, because I do not understand that part good enough to do it without code duplication. But at least, the code duplication is obvious in the code, and next to each other. I hope that helps to get rid of the code duplication easily.

How should I rename `tySequence`?

**Comment by Araq** [comment]
Closing since we merged some other PR improving gettype() tremendously.

**Pull request closed without merging**